tests/test_scheduler/test_unit/test_base.py

```diff
@@ -76,18 +65,25 @@ def test_try_allocation(self, mocked_init, mocked_schedule_unit, mocked_handle_c
         component._wait_pool = list()
         component._wait_lock = threading.RLock()
         component._slot_lock = threading.RLock()
-        unit = {'description':{'note':'this is a unit'}, 'uid': 'test'}
-        component._try_allocation(unit=unit)
-        self.assertEqual(unit['slots'], {"cores_per_node": 16,
-                                 "lfs_per_node": {"size": 0, "path": "/dev/null"},
-                                 "nodes": [{"lfs": {"path": "/dev/null", "size": 0},
-                                            "core_map": [[0]],
-                                            "name": "a",
-                                            "gpu_map": None,
-                                            "uid": 1,"mem": None}],
-                                 "lm_info": "INFO",
-                                 "gpus_per_node": 6,
-                                 })
+
+        tests = self.setUp()['try_allocation']
+        for input_data, result in zip(tests['setup'], tests['results']):
+            component.schedule_unit = mock.Mock(
+                return_value=input_data['scheduled_unit_slots'])
+
+            unit = input_data['unit']
+            component._try_allocation(unit=unit)
+
+            # test unit's slots
+            self.assertEqual(unit['slots'], result['slots'])
+
+            # test environment variable(s)
+            if 'NODE_LFS_PATH' not in result['description']['environment']:
+                self.assertNotIn('NODE_LFS_PATH',
+                                 unit['description']['environment'])
```

Comment:
I think that the following `assertEqual` will fail if `NODE_LFS_PATH` is not in the result and is in the unit's environment. Am I missing something?

Having said that I am perfectly fine with leaving it.

Reply:
that's correct, it should, since in `result` is our expectations, which should reflect the state of `unit`. There are 2 test cases, in one of them this env variable is expected to be set and in another one this env variable should not be in unit's `environment`

Reply:
If that is the case, does removing it make sense to you?

Reply:
sorry, can you explain more what should be removed?

Reply:
I mean the `if... else` statement. The check in line 85 should be more than enough, right?

Reply:
yeah, you are right, the next one `assertEqual` duplicates this one, I thought to emphasize this env variable check, but the following assert would be enough. Will update it accordingly